### ISSUE TITLE
create private definition should NOT allow public object instantiation 

### ISSUE THREAD

**Issue by developedbysom** [abaplint/abaplint#2356]
Hi,
I am not sure if this is already a known issue. If I create a class definition as private, I shouldn't be allowed to create an instantiation outside the class static method.

```abap
class lcl_test definition create private.
   public section.
   methods get_data returning value(re_data) type string.
endclass.

class lcl_test implementation.
  method get_data.
  re_data = 'Hello'.
endmethod.
endclass.

start-of-selection.
data lo_obj_instance type ref to lcl_test.
*-------->> below object instantiation should give a syntax error, as class is defined as private.
create object lo_obj_instance.
lo_obj_instance->get_data( ). // This should give syntax error.
```

but I see in abaptranspiler this is accepting and so calling out as an issue.

-- Thanks, Somnath

**Comment by larshp** [comment]
@developedbysom try again, latest version
